### PR TITLE
Feat: Support methods for miner as coordinator

### PR DESCRIPTION
**Pull request by kantai**
This adds a few methods for supporting the miner as coordinator:

1. Adds `set_key_and_party_polynomials()` which sets the coordinator up with the given parameters, checking that they match.
2. Adds some friendlier debug fmt strings.
3. Marks a few fields as pub as necessary from the stacks-node nakamoto miner.

### CODE REVIEW

File: src/state_machine/coordinator/fire.rs [hunk 1]
```diff
@@ -661,6 +662,33 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
         Ok(())
     }
 
+    /// Set the aggregate key and polynomial commitments used to form that key.
+    ///  Check if the polynomial commitments match the key
+    pub fn set_key_and_party_polynomials(
+        &mut self,
+        aggregate_key: Point,
+        party_polynomials: Vec<(u32, PolyCommitment)>,
+    ) -> Result<(), String> {
```
Comment (netrome) [inline src/state_machine/coordinator/fire.rs:671]: Nit: I would prefer to see an enum rather than a string as the error type here.

**Review by netrome — APPROVED** [review]
Just one minor nit.

File: src/state_machine/coordinator/fire.rs (hunk 1 above)
Comment (kantai) [reply src/state_machine/coordinator/fire.rs]: Yep, updated.

Comment (netrome) [reply src/state_machine/coordinator/fire.rs]: Nice, thanks!

**Review by xoloki — APPROVED** [review]
LGTM just a quick question about hex vs base58

File: src/state_machine/coordinator/mod.rs [hunk 2]
```diff
@@ -74,6 +74,14 @@ pub enum Error {
     /// DKG failure from signers
     #[error("DKG failure from signers")]
     DkgFailure(HashMap<u32, DkgFailure>),
+    /// Aggregate key does not match supplied party polynomial
+    #[error(
+        "Aggregate key and computed key from party polynomials mismatch: got {0}, expected {1}"
+    )]
+    AggregateKeyPolynomialMismatch(Point, Point),
```
Comment (xoloki) [inline src/state_machine/coordinator/mod.rs:81]: `clippy` doesn't like the size of this variant so I'm gonna squelch the error

Comment (xoloki) [reply src/state_machine/coordinator/mod.rs:82]: Fixed

File: src/net.rs [hunk 3]
```diff
@@ -260,6 +262,19 @@ pub struct NonceRequest {
     pub merkle_root: Option<MerkleRoot>,
 }
 
+impl Debug for NonceRequest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("NonceRequest")
+            .field("dkg_id", &self.dkg_id)
+            .field("sign_id", &self.sign_id)
+            .field("sign_iter_id", &self.sign_iter_id)
+            .field("message", &hex::encode(&self.message))
```
[the remark this answers is not in the document]
Comment (kantai) [reply src/net.rs:271]: I generally find hex encoding easier to read for things like message serializations: these messages end up containing things like the block hash for a proposal, and those are generally printed in hex elsewhere.

**Comment by xoloki** [comment]
Are you ready for me to merge this @kantai ?

**Comment by kantai** [comment]
Yep, let's do it

**Pull request merged**